api/v1alpha1/httpsedge_types.go

```diff
@@ -68,6 +68,9 @@ type HTTPSEdgeSpec struct {
 
 	// Routes is a list of routes served by this edge
 	Routes []HTTPSEdgeRouteSpec `json:"routes,omitempty"`
+
+	// TLSTermination is the TLS termination configuration for this edge
+	TLSTermination *EndpointTLSTerminationAtEdge `json:"tlsTermination,omitempty"`
```

Comment:
I know on the others modules we removed the "enabled" attribute in favor of just enabling it if there is a value present. This one though we could keep the enabled as without it the functionality changes a bit. If we had both enabled and min_version, then based on the API, the min_version could be optional and would default`if unspecified, ngrok will choose an industry-safe default`.

relying on that though does make the api a bit more confusing in that if you set the min_version it won't do anything (or would error) unless you also set enabled

Reply:
From the [docs](https://ngrok.com/docs/cloud-edge/modules/tls-termination) page:

> This module is used to configure how the TLS connection is terminated at the edge. For HTTPS Edges, TLS is always terminated at the ngrok Edge which means you can only configure the TLS version enforced there. For TLS tunnels, you can configure where the TLS termination occurs.

> The default TLS version for ngrok is 1.2. You can configure the minimum TLS Version via the TLS module. The minimum TLS version ngrok supports is 1.0 up to the latest version 1.3.

it looks like for HTTPSEdges, TLS is always terminated at the ngrok Edge. Exposing `Enabled` seems extraneous to me in a configuration as code(kubernetes manifests). If values are supplied in the manifests, we'll want it enabled. If they are not supplied, the module is disabled/removed. Open to adding this back in though if there is a compelling reason.

Reply:
Ahh, I misunderstood the docs. I didn't look at the others to see enabled was on them all (i thought it was something we had done, not from the api). The fact that its always on makes sense